core/src/main/java/com/linecorp/armeria/client/HttpChannelPool.java

```diff
@@ -204,8 +217,18 @@ private static <T> T[] newEnumMap(Class<?> elementType,
         return maps;
     }
 
-    private Bootstrap getBootstrap(SessionProtocol desiredProtocol) {
-        return bootstraps[desiredProtocol.ordinal()];
+    private Bootstrap getBootstrap(SessionProtocol desiredProtocol, SocketAddress remoteAddress) {
+        if (remoteAddress instanceof InetSocketAddress) {
+            return inetBootstraps[desiredProtocol.ordinal()];
+        }
+
+        assert remoteAddress instanceof DomainSocketAddress : remoteAddress;
+
+        if (unixBootstraps == null) {
+            throw new UnsupportedAddressTypeException();
```

Comment:
Optional) What do you think of providing a more verbose exception here since a user might not know why domain sockets are not supported?

```suggestion
            throw new IllegalArgumentException("Domain sockets are not supported by " + eventLoop.parent().toString());
```

Reply:
That's a good idea. It's a shame that `UnsupportedAddressTypeException` doesn't accept a message.